### PR TITLE
Adding an alternative formulation of H-levels

### PR DESCRIPTION
**Pull request by IanRay11**
I also added my name to the read me file here. Hope that is okay.

### CODE REVIEW

File: source/UF/H-Levels.lagda [hunk 1]
```diff
@@ -0,0 +1,163 @@
+Martin Escardo and Ian Ray, 06/02/2024
```
Comment (martinescardo) [inline source/UF/H-Levels.lagda:1]: This should be "Ian Ray" only, I think.

**Review by martinescardo — CHANGES_REQUESTED** [review]
Minimal change requested.

File: source/UF/H-Levels.lagda (hunk 1 above)
Comment (IanRay11) [reply source/UF/H-Levels.lagda:1]: I copied some of your proofs from the original file :)

**Comment by IanRay11** [comment]
I am actually currently working on showing H n is of level (n + 1) then also, then we can merge. Unless you have more suggestions

**Pull request merged**